src/hist/core.py

```diff
@@ -34,13 +48,40 @@ def __init__(self, *args, **kwargs):
 
         super().__init__(*args, **kwargs)
         self.names: dict = dict()
-        for ax in self.axes:
-            if ax.name in self.names:
-                raise Exception(
-                    f"{self.__class__.__name__} instance cannot contain axes with duplicated names."
-                )
-            else:
-                self.names[ax.name] = True
+        if hasattr(self, axes):
+            for ax in self.axes:
+                if ax.name in self.names:
+                    raise Exception(
+                        f"{self.__class__.__name__} instance cannot contain axes with duplicated names."
+                    )
+                else:
+                    self.names[ax.name] = True
+        
+        self._hist = None
+        self._ax = []
+        self._storage_proxy = None
+    
+    @always_normal_method
+    def Regular(self):
+        if self._hist:
+            raise RuntimeError("Cannot add an axis to an existing histogram")
+        self._ax.append("Regular")
+        return self
+
+    @always_normal_method
+    def Variable(self):
+        if self._hist:
+            raise RuntimeError("Cannot add an axis to an existing histogram")
+        self._ax.append("Variable")
+        return self
+
+    def __getattribute__(self, item):
+        if not self._hist and not isinstance(getattr(self.__class__, item), always_normal_method):
```

Comment:
`getattr(self, ...)` calls `self.__getattribute__(...)`, which is this function again - so this is recursive. I think you need `object.__getattribute__(self, item)`.

Reply:
Oops, even sooner,`self._hist` is calling `__getattribute__`! Here's a version that should get you past this:

```python
    def __getattribute__(self, item):
        if not super().__getattribute__("_hist") and not isinstance(super().__getattribute__(item), always_normal_method):
            # Make histogram real here
            super().__init__(*self._ax, storage=self._storage_proxy)
            self._storage_proxy = None

        return super().__getattribute__(item)
```

Reply:
> Here's a version that should get you past this:

Sadly, it cannot, still a recursive in the code :'(. Let me see whether I can figure it out.